File: 01_Language/01_Functions/python/flock.py
# coding: utf-8

import fcntl

if __name__ == '__main__':
    f = open("test.txt", "r+")
    fcntl.flock(f, fcntl.LOCK_EX)
    f.truncate(0)
    f.write("hello world")
    f.flush()
    fcntl.flock(f, fcntl.LOCK_UN)
    f.close()

File: 01_Language/01_Functions/php/flock.php
<?php

$fp = fopen("test.txt", "r+");

if (flock($fp, LOCK_EX)) {  // acquire an exclusive lock
    ftruncate($fp, 0);      // truncate file
    fwrite($fp, "Write something here\n");
    fflush($fp);            // flush output before releasing the lock
    flock($fp, LOCK_UN);    // release the lock
} else {
    echo "Couldn't get the lock!";
}

fclose($fp);
